docs/src/main/asciidoc/kotlin.adoc

```diff
@@ -275,38 +283,106 @@ dependencies {
-compileTestKotlin.kotlinOptions {
-    jvmTarget = "1.8"
-}
+<1> The Kotlin plugin version needs to be specified.
+<2> The all-open configuration required, as per Maven guide above
+
+
+
+== Live reload
+
+Quarkus provides support for live reloading changes made to source code. This support is also available to Kotlin, meaning that developers can update their Kotlin source
+code and immediately see their changes reflected.
+
+To see this feature in action, first execute: `./mvnw compile quarkus:dev`
+
+When executing an HTTP GET request against `http://localhost:8080/greeting`, you see a JSON message with the value `hello` as its `message` field.
+
+Now using your favorite editor or IDE, update `GreetingResource.kt` and change the `hello` method to the following:
+
+[source,kotlin]
+----
+fun hello() = Greeting("hi")
+----
+
+When you now execute an HTTP GET request against `http://localhost:8080/greeting`, you should see a JSON message with the value `hi` as its `message` field.
+
+One thing to note is that the live reload feature is not available when making changes to both Java and Kotlin source that have dependencies on each other. We hope to alleviate this limitation in the future.
+
+== Packaging the application
+
+As usual, the application can be packaged using `./mvnw clean package` and executed using the `target/quarkus-app/quarkus-run.jar` file. You can also build the native executable using `./mvnw package -Pnative`, or `./gradlew buildNative`.
+
+== Kotlin and Jackson
+
+If the `com.fasterxml.jackson.module:jackson-module-kotlin` dependency and the `quarkus-jackson` extension (or the `quarkus-resteasy-jackson` extension) have been added to the project,
+then Quarkus automatically registers the `KotlinModule` to the `ObjectMapper` bean (see link:rest-json#jackson[this] guide for more details).
+
+When using Kotlin data classes with `native-image` you may experience serialization errors that do not occur with the `JVM` version, despite the Kotlin Jackson Module being registered. This is especially so if you have a more complex JSON hierarchy, where an issue on a lower node causes a serialization failure. The error message displayed is a catch-all and typically displays an issue with the root object, which may not necessarily be the case.
+
+[source]
+----
+com.fasterxml.jackson.databind.exc.InvalidDefinitionException: Cannot construct instance of `Address` (no Creators, like default construct, exist): cannot deserialize from Object value (no delegate- or property-based Creator)
+----
+
+To ensure full-compability with `native-image`, it is recommended to apply the Jackson `@field:JsonProperty("fieldName")` annotation, and set a nullable default, as illustrated below. When using JSON to Kotlin data class auto-generators, such as those found in IntelliJ, you can typically configure these options without any extra burden, i.e. enable the Jackson annotation and use nullable parameters as part of the auto-code generation.
```

Comment:
Something is off here "When using JSON to Kotlin data class auto-generators".  Should "serialize" go before Kotlin?

Reply:
I'm actually referring here to the code generator tools, I'll make it more obvious (not talking about serialisation, but code generator from JSON-->data class).

Reply:
Changed the sentence to:

```
You can automate the generation of Kotlin data classes for your sample JSON using Intellij plugins (such as JSON to Kotlin Class), and easily enable the Jackson annotation and select nullable parameters as part of the auto-code generation.
```